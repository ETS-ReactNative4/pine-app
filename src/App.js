import React, { Component } from 'react';
import { View } from 'react-native';
import { Provider } from 'react-redux';

import { handle as handleError } from './actions/error';
import AppNavigator from './navigators/AppNavigator';
import ServiceManager from './services/ServiceManager';
import { ThemeProvider } from './contexts/theme';
import getAppWithNavigationState from './getAppWithNavigationState';
import createStore from './createStore';

const navReducer = (state, action) => {
  const nextState = AppNavigator.router.getStateForAction(action, state);
  return nextState || state;
};

const store = createStore(navReducer);
const AppWithNavigationState = getAppWithNavigationState();

export default class App extends Component {
  constructor() {
    super(...arguments);
    this._services = new ServiceManager(store);
  }

  componentDidMount() {
    try {
      this._services.start();
    } catch (error) {
      store.dispatch(handleError(error));
    }
  }

  componentWillUnmount() {
    this._services.stop();
  }

  render() {
    return (
      <Provider store={store}>
        <ThemeProvider>
          <View style={{ flex: 1, alignSelf: 'stretch' }}>
            <AppWithNavigationState />
          </View>
        </ThemeProvider>
      </Provider>
    );
  }
}
