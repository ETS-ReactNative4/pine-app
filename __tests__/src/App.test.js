import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../src/App';

jest.mock('../../src/screens/WelcomeScreen', () => 'WelcomeScreen');

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <App />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
