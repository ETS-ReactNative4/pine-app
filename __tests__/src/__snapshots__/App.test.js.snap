// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "flex": 1,
            "flexDirection": "column-reverse",
            "overflow": "hidden",
          },
          Object {
            "backgroundColor": "#000",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "#E9E9EF",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "paddingTop": 0,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.2,
              "shadowRadius": 5,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              onMoveShouldSetResponder={[Function]}
              onMoveShouldSetResponderCapture={[Function]}
              onResponderEnd={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderReject={[Function]}
              onResponderRelease={[Function]}
              onResponderStart={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              onStartShouldSetResponderCapture={[Function]}
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column-reverse",
                    "overflow": "hidden",
                  },
                  Object {
                    "backgroundColor": "#000",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  pointerEvents="auto"
                  style={
                    Object {
                      "backgroundColor": "#E9E9EF",
                      "bottom": 0,
                      "left": 0,
                      "opacity": 1,
                      "paddingTop": 0,
                      "position": "absolute",
                      "right": 0,
                      "shadowColor": "black",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0.2,
                      "shadowRadius": 5,
                      "top": 0,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                      ],
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                        "overflow": "hidden",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#ffffff",
                              "flex": 1,
                              "justifyContent": "center",
                              "padding": 40,
                            },
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "#FFD200",
                              "flex": 1,
                              "justifyContent": "center",
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignSelf": "stretch",
                              "backgroundColor": "transparent",
                              "height": 44,
                              "justifyContent": "center",
                              "left": 0,
                              "marginTop": 20,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "black",
                                  "fontFamily": "System",
                                  "fontSize": 17,
                                  "fontWeight": "600",
                                },
                                Object {
                                  "left": 40,
                                  "position": "absolute",
                                  "right": 40,
                                  "textAlign": "center",
                                },
                              ]
                            }
                          />
                        </View>
                        <Image
                          source={
                            Object {
                              "uri": "LaunchScreenLogo",
                            }
                          }
                          style={
                            Object {
                              "height": 57,
                              "width": 128,
                            }
                          }
                        />
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "backgroundColor": "#ffffff",
                                "bottom": 0,
                                "left": 40,
                                "paddingBottom": 30,
                                "position": "absolute",
                                "right": 40,
                              },
                              null,
                              Object {
                                "backgroundColor": "transparent",
                              },
                            ]
                          }
                        >
                          <ActivityIndicator
                            animating={true}
                            color="#FFFFFF"
                            hidesWhenStopped={true}
                            size="small"
                            style={
                              Object {
                                "height": 42,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    pointerEvents="none"
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 2000,
      }
    }
  >
    <View
      blurAmount={5}
      blurType="dark"
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "opacity": 0,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#ffffff",
            "borderRadius": 12,
            "marginTop": 0,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 12,
            "width": 710,
          },
          Object {
            "marginTop": -500,
            "opacity": 0,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFD6D6",
            "borderTopLeftRadius": 12,
            "borderTopRightRadius": 12,
            "padding": 15,
            "paddingTop": 17,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "color": "#FF5A59",
                "fontSize": 16,
                "fontWeight": "600",
                "textAlign": "center",
              },
            ]
          }
        >
          <IoniconIcon
            name="ios-alert"
            style={
              Object {
                "fontSize": 28,
              }
            }
          />
        </Text>
      </View>
      <View
        style={
          Object {
            "padding": 25,
            "paddingLeft": 30,
            "paddingRight": 30,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Array [
                Object {
                  "color": "#B1AFB7",
                  "fontSize": 15,
                  "fontWeight": "400",
                  "letterSpacing": -0.24,
                  "lineHeight": 21,
                  "marginBottom": 20,
                },
                Object {
                  "marginBottom": 0,
                },
              ],
            ]
          }
        >
          
        </Text>
      </View>
      <View>
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderBottomLeftRadius": 12,
              "borderBottomRightRadius": 12,
              "borderColor": "#fbfbfb",
              "borderTopWidth": 1,
              "padding": 15,
            }
          }
        >
          <Text
            style={null}
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#322A51",
                    "fontWeight": "600",
                    "letterSpacing": 1,
                  },
                ]
              }
            >
              OK
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
